parser-typechecker/src/Unison/Server/Backend.hs

```diff
@@ -148,7 +151,9 @@ basicNames' root scope =
         externalNames = rootNames `Names.difference` pathPrefixed currentPathNames
         rootNames = Branch.toNames root0
         pathPrefixed = case path of
-          Path.Path (toList -> []) -> const mempty
+          -- TODO: This seems like a bug, if we have an empty path, we always return empty
+          -- names!
+          Path.Empty -> const mempty
```

Comment:
@hojberg  I noticed this, I think this might be related to some oddities we're seeing.